### PR TITLE
docstore: remove compile-time shaded protobuf dep from Ali canonicalizer

### PR DESCRIPTION
**Pull request by kchoy-sfdc**
## Summary
`TablestoreBodyCanonicalizer.appendRow` previously declared a `ByteString` parameter (from the Tablestore SDK's shaded protobuf: `com.aliyun.ots.thirdparty.com.google.protobuf.ByteString`), creating a compile-time dependency on the shaded jar. This change removes it: `appendRow` now accepts `byte[]` instead, with `.toByteArray()` called inline at the two call sites (`req.getRow().toByteArray()`, `req.getPrimaryKey().toByteArray()`). Java does not require importing a type that is only used transiently as an intermediate return value, so the shaded jar is no longer referenced at compile time — only the existing runtime dependency via the SDK's internal implementation remains.

## Motivation
The Tablestore SDK's shaded protobuf artifacts have independent versioning from the main SDK jar. Removing the compile-time reference decouples the canonicalizer from that versioning and avoids the need to explicitly manage the shaded jar as a build dependency.

## Testing
Expanded `TablestoreBodyCanonicalizerTest` from 4 to 7 tests, adding coverage for:
- DeleteRow path and PK-only PlainBuffer parsing
- Condition field presence and revision-epoch disambiguation (bare vs. conditional delete)
- Malformed body (missing mandatory fields) returns original bytes unchanged

All 7 tests pass. `AliDocstoreIT` 8/8 pass in no-credentials replay mode.

## Cross-cloud compatibility
Test-harness-only change (`src/test`). No production code modified.

### CODE REVIEW

File: docstore/docstore-ali/src/test/java/com/salesforce/multicloudj/docstore/ali/TablestoreBodyCanonicalizer.java [hunk 1]
```diff
@@ -65,36 +66,70 @@ public static byte[] canonicalize(String url, byte[] body) {
...
   }
 
-  private static void appendRow(ObjectNode root, ByteString rowBytes) throws Exception {
+  private static void appendRow(ObjectNode root, byte[] rowBytes) throws Exception {
     PlainBufferCodedInputStream in =
-        new PlainBufferCodedInputStream(new PlainBufferInputStream(rowBytes.toByteArray()));
+        new PlainBufferCodedInputStream(new PlainBufferInputStream(rowBytes));
     List<PlainBufferRow> rows = in.readRowsWithHeader();
 
     // A write request carries a single row; represent PK and cells as keyed objects so
```
Comment (iamabhilaksh) [inline docstore/docstore-ali/src/test/java/com/salesforce/multicloudj/docstore/ali/TablestoreBodyCanonicalizer.java:104]: Nice touch pulling the condition straight off the wire, but I want to sanity-check one equivalence: the old path emitted `base16(req.getCondition().toByteArray())`, which re-serializes the parsed sub-message, whereas we now hex the raw field-3 wire bytes from `readByteArray()`. For live record/replay that's fine since both sides run this same path — but if any committed fixture was canonicalized under the old re-serialized form, its stored `condition` hex could differ from what replay now produces and quietly stop matching. Did we regenerate/confirm the existing Tablestore stubs against this new path, or is condition-bearing coverage new enough that there's nothing stale to break?

File: docstore/docstore-ali/src/test/java/com/salesforce/multicloudj/docstore/ali/TablestoreBodyCanonicalizerTest.java [hunk 2]
```diff
@@ -127,10 +131,107 @@ void recordAndReplayPairingMatchesAcrossColumnOrder() {
...
+  void deleteRowCanonicalizesCorrectly() {
+    // A DeleteRow carries only primary-key cells (no data cells). Verify the canonicalizer extracts
+    // table + pk correctly and produces a valid canonical JSON.
+    byte[] body = deleteBody("docstore_test_2", "Game", "test-game", null);
+
+    String canonical =
+        new String(
+            TablestoreBodyCanonicalizer.canonicalize("/DeleteRow", body), StandardCharsets.UTF_8);
+
+    assertTrue(canonical.contains("\"DeleteRow\""), "op must be DeleteRow");
```
Comment (iamabhilaksh) [inline docstore/docstore-ali/src/test/java/com/salesforce/multicloudj/docstore/ali/TablestoreBodyCanonicalizerTest.java]: `deleteRowCanonicalizesCorrectly` here and `conditionFieldDistinguishesRevisions` at L218 assert with substring `canonical.contains(...)`, while the rest of this file pins structure via `equalToJson`/`assertArrayEquals`. The substring form still passes if a value lands in the wrong place — `table` ending up under `rows[0].pk`, or an unexpected extra field next to `condition`. Since this canonicalizer's whole job is an exact stable structure for WireMock's `equalToJson`, could we tighten these two to a structural assertion (even just parsing into a `JsonNode` and checking specific paths)? Happy to be wrong if the looseness is deliberate.

File: docstore/docstore-ali/src/test/java/com/salesforce/multicloudj/docstore/ali/TablestoreBodyCanonicalizer.java (hunk 1 above)
Comment (kchoy-sfdc) [reply docstore/docstore-ali/src/test/java/com/salesforce/multicloudj/docstore/ali/TablestoreBodyCanonicalizer.java:104]: Good catch on the equivalence question. `Condition` is a proto2 *message* type (extends `GeneratedMessage`), so the old path re-serialized the parsed sub-message via `.toByteArray()`, while the new path takes the raw field-3 wire bytes. For the existing stubs these produce identical hex: proto2 re-serialization of a message with no unknown fields is byte-for-byte the same as the original wire encoding (field order preserved, no normalization). All committed condition values use only the standard `row_existence` and `column_condition` fields with no unknown fields. Confirmed empirically — `AliDocstoreIT` 8/8 passes in no-creds replay including the revision-conditional stubs.

The two paths would diverge only if the `Condition` message on the wire carried unknown fields — which happens when the SDK version used to *write* the request knows about a field that the version used to *parse* does not. In practice this cannot occur here: record and replay always run against the same SDK version (pom-pinned), and `Condition` is a stable core wire-protocol type that has not gained new fields across the SDK versions in scope. If the SDK ever does evolve `Condition`, the stubs would need re-recording anyway (request bodies change), and the pre-PR no-creds replay gate would catch the mismatch before it reaches CI.

File: docstore/docstore-ali/src/test/java/com/salesforce/multicloudj/docstore/ali/TablestoreBodyCanonicalizerTest.java (hunk 2 above)
Comment (kchoy-sfdc) [reply docstore/docstore-ali/src/test/java/com/salesforce/multicloudj/docstore/ali/TablestoreBodyCanonicalizerTest.java]: Good point — tightened in the follow-up commit. `deleteRowCanonicalizesCorrectly` now parses the canonical JSON and asserts exact field paths (`root.get("op")`, `root.get("table")`, `rows[0].pk.Game.v`), and `conditionFieldDistinguishesRevisions` checks `path("condition").isMissingNode()` and `asText().isEmpty()` directly on the `JsonNode`. No more substring matching.

File: docstore/docstore-ali/src/test/java/com/salesforce/multicloudj/docstore/ali/TablestoreBodyCanonicalizer.java [hunk 3]
```diff
@@ -65,36 +66,70 @@ public static byte[] canonicalize(String url, byte[] body) {
...
+          : OtsInternalApi.DeleteRowRequest.TABLE_NAME_FIELD_NUMBER;
+
+      String tableName = null;
+      byte[] rowBytes = null;
+      byte[] conditionBytes = null;
+
+      CodedInputStream cin = CodedInputStream.newInstance(body);
+      int tag;
+      while ((tag = cin.readTag()) != 0) {
+        int fieldNumber = WireFormat.getTagFieldNumber(tag);
```
Comment (LihaoLiuXs) [inline docstore/docstore-ali/src/test/java/com/salesforce/multicloudj/docstore/ali/TablestoreBodyCanonicalizer.java]: The generated protobuf parser dispatches on the complete tag (10/18/26), but this loop matches only the field number. That changes behavior for a known field encoded with the wrong wire type: the generated parser treats it as unknown and can continue to a later valid occurrence, while this code calls `readString`/`readByteArray` against the wrong type and consumes unrelated bytes or falls back to the raw body. I reproduced this by prepending varint tag 8/value 1 to a valid PutRow: `OtsInternalApi.PutRowRequest.parseFrom` still reads the correct table, but this canonicalizer returns non-JSON raw bytes. `skipField(tag)` can also return false for an end-group and that result is currently ignored. Please match the exact length-delimited tags (or validate `WireFormat.getTagWireType(tag)`) and return the original body when `skipField` returns false; a wrong-wire known-field test would pin the old parser semantics.

Comment (kchoy-sfdc) [reply docstore/docstore-ali/src/test/java/com/salesforce/multicloudj/docstore/ali/TablestoreBodyCanonicalizer.java]: Good points — rather than hardening the hand-rolled wire parser to match the generated code's edge-case semantics, we simplified back to `parseFrom`. The one change from the original is that `appendRow` now accepts `byte[]` instead of `ByteString`, with `.toByteArray()` called inline at the call sites (`req.getRow().toByteArray()`, `req.getPrimaryKey().toByteArray()`). Java does not require importing a type that is only used transiently as an intermediate return value, so there is no compile-time reference to the shaded jar — only the existing runtime one. The generated parser handles all wire-format edge cases (wrong wire types, unknown fields, end-group markers) exactly as the spec requires.

**Review by LihaoLiuXs — APPROVED** [review]
